### ISSUE TITLE
Rule Change: Report statically invalid radix values in unary expressions

### ISSUE THREAD

**Issue by Pixel998** [eslint/eslint#21046]
### What rule do you want to change?

radix

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint radix: "error" */

parseInt("10", +"37");
Number.parseInt("10", +"37");
```

### What does the rule currently do for this code?

The rule does not report either call, even though `+"37"` is statically known to evaluate to `37`, which is not a valid radix.

### What will the rule do after it's changed?

The rule reports these calls because the radix argument can be statically determined to evaluate to an invalid radix value.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

https://github.com/eslint/eslint/pull/21030/changes#r3503951717

**Comment by DMartens** [comment]
I think if we add support, we should not check for everything which would not be a valid radix.
Your examples only include string literals but there are other examples which can be statically detected (e.g. `object?.prop` -> may be undefined which the rule forbids).
An alternative would be a `strict` option which only allows numbers which can be statically resolved to be a valid radix.
